### ISSUE TITLE
JSXC RELOGIN ISSUE

### ISSUE THREAD

**Issue by sunstar20** [jsxc/jsxc#695]
### Actual behavior: Strophe is getting disconnected at times and as a result, the relogin link is shown.

![image](https://user-images.githubusercontent.com/12511931/38491865-c2b601e6-3c0a-11e8-8148-32e5e294d2f4.png)

### Steps to reproduce the behavior
1. Reload the page which has integrated the agent chat

### Environment
- **JSXC version:** 3.3.1
- **Host system and version:** 
- **Browser vendor and version:**  Chrome
- **Any browser plugins enabled?** NO
- **XMPP server vendor and version:**  ejabberd 17.09
- **Is your XMPP server working with other clients as expected?** NA

#### Javascript

``` NO

#### JSXC
<!-- Which messages appear in your javascript console if you enable debugging? -->
<!-- How to enable debugging: https://github.com/jsxc/jsxc/wiki/Debugging -->
```

#### XMPP

``` NO

#### Host
<!-- Do you have any related entries in your host system log? -->
```

**Comment by sualko** [comment]
The js log would be helpful, but in the meantime try to increase your bosh inactive time.

**Comment by sunstar20** [comment]
Adding my Analysis: Please correct me if I am wrong.

On analyzing the scenario using console logs, the above scenario occurs when the strophe connection attempt has failed i.e CONNFAIL.

    case Strophe.Status.CONNFAIL:
            console.log("CONNFAIL");
            $(document).trigger('connfail.jsxc', condition);
            break;
When the connection attempt has failed, connfail.jsxc will be triggered and will execute the function 'onConnfail'.
    
$(document).on('connfail.jsxc', jsxc.xmpp.onConnfail);
   
 /**
    * Triggered on connection fault
    *
    * @param {String} condition information why we lost the connection
    * @private
    */
    onConnfail: function(ev, condition) {
      console.log("ev : "+ev+" and condition : "+condition);
      jsxc.debug('XMPP connection failed: ' + condition);
      if (jsxc.options.loginForm.triggered) {
        console.log("jsxc.options.loginForm.triggered true");
         jsxc.submitLoginForm();
      }
   },

The condition for connection failure will be shown as below:
ev : [object Object] and condition : item-not-found

When the Strophe connection fails, Strophe connection will terminate/disconnect.
case Strophe.Status.DISCONNECTED:
            console.log("DISCONNECTED");
            $(document).trigger('disconnected.jsxc');
            break;

CONSOLE LOG:

![image](https://user-images.githubusercontent.com/12511931/38596561-da93fa96-3d6f-11e8-89a2-3850eca28dc6.png)

**Comment by sunstar20** [comment]
Could you please confirm whether the bosh inactive time should be set in the config file (yml file)?

**Comment by sunstar20** [comment]
I couldnt locate the field in ejabberd.yml file. could you please help

**Comment by MarcelWaldvogel** [comment]
See https://docs.ejabberd.im/admin/configuration/#options-1

**Comment by sunstar20** [comment]
Whenever the relogin issue occurs, the below error is shown: can you please help

![relogin](https://user-images.githubusercontent.com/12511931/40827735-21a973da-659c-11e8-9906-53165436e0f7.png)

**Comment by sunstar20** [comment]
@MarcelWaldvogel 

Tried adding the max_inactivity but the issue still occurs intermittently

![image](https://user-images.githubusercontent.com/12511931/40828034-20c18402-659d-11e8-9d62-2bb192cc5fe9.png)

**Comment by sunstar20** [comment]
PFA the jsxc.js file that I'm using. Kindly help.

[jsxc_js.txt](https://github.com/jsxc/jsxc/files/2061619/jsxc_js.txt)

**Comment by MarcelWaldvogel** [comment]
@sualko ?

**Comment by sualko** [comment]
Why did you attach the jsxc file? Did you modify the file? Can you try the latest release? Can you show your jsxc init and start code?

**Comment by anishtakkar7** [comment]
@sualko i am going through same issue. Can you give me the actual solution to this problem. When I login,  after sometime ( around 120 sec) it disconnects. Please explain how to keep it alive forever

**Comment by sualko** [comment]
> Please explain how to keep it alive forever

I never experienced the described scenario and we never got such reports in our university setup. Therefore I have no glue why this happens to you and without a error message I can't do anything. Sorry. If your installation is public available you can send some test credentials to klaus@jsxc.org and I will look at it.

**Comment by anishtakkar7** [comment]
You have jsxc working example here :  https://github.com/jsxc/jsxc
I was run this code on my machine and I saw after some amount of idle time the chatlist disappears and it shows No connection relogin, 
Following below are the snapshots when chat is working properly : 
1. ![jsxc workign](https://user-images.githubusercontent.com/39048023/44298038-82370380-a2f9-11e8-9c5f-0951a4b1225b.png)

2.
![jsxc working 2](https://user-images.githubusercontent.com/39048023/44298040-8e22c580-a2f9-11e8-80bd-b9918433be6a.png)


Following below the snapshot when chat lost the connection : 

![jsxc not working](https://user-images.githubusercontent.com/39048023/44298051-c1655480-a2f9-11e8-9c1c-7fe63400dde7.png)


I have downloaded code from the above provided github link of this example, try to replicate this issue and tell me where I need to make changes.
